core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -169,6 +172,32 @@ class CloudSqlInstance {
     }
   }
 
+  private OAuth2Credentials parseCredentials(HttpRequestInitializer source) {
```

Comment:
Can you add a simple unit test for this method which tests all 3 cases (HttpCredentialsAdapter, Credential, and other unsupported type)

Reply:
Will have a look at it.